org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/PresetManagerPage.java

```diff
@@ -58,29 +86,191 @@ public PresetManagerPage(PresetRepository repository) {
 
 	@Override
 	public void createControl(Composite parent) {
-		initializeDialogUnits(parent);
-
 		setTitle(MESSAGE_PRESET_MANAGER_PAGE_TITLE);
 		setDescription(MESSAGE_PRESET_MANAGER_PAGE_DESCRIPTION);
 
 		ScrolledComposite sc = new ScrolledComposite(parent, SWT.H_SCROLL | SWT.V_SCROLL);
 		Composite container = new Composite(sc, SWT.NONE);
 		sc.setContent(container);
+		container.setLayout(new GridLayout(2, false));
 
-		// TODO: create manager control here
-		container.setLayout(new FillLayout());
-		new Label(container, SWT.NONE).setText("TODO: create manager control here");
+		createPresetTable(container);
+		createButtons(container);
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
+
+		bindListeners();
+	}
+
+	private void createPresetTable(Composite parent) {
+		tableViewer = new TableViewer(parent, SWT.V_SCROLL | SWT.BORDER | SWT.MULTI);
+		tableViewer.getControl().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		tableViewer.setContentProvider(new PresetTableContentProvider());
+		tableViewer.setLabelProvider(new DelegatingStyledCellLabelProvider(new PresetTableLabelProvider()));
+		tableViewer.setInput(repository);
+	}
+
+	private void createButtons(Composite parent) {
+		Composite container = new Composite(parent, SWT.NONE);
+		container.setLayoutData(new GridData(SWT.FILL, SWT.BEGINNING, false, true));
+		GridLayout layout = new GridLayout(1, true);
+		layout.marginHeight = 0;
+		layout.marginWidth = 0;
+		container.setLayout(layout);
+
+		newButton = createButton(container, MESSAGE_NEW_BUTTON);
```

Comment:
perhaps we can move this button functionality to the tableButtonsControl class because it seems like these actions are being reused in a bunch of places?

Reply:
`TableInspector` is refactored to handle buttons as well.